sanic/router.py

```diff
@@ -163,10 +163,10 @@ def add_parameter(match):
             parameters.append(parameter)
 
             # Mark the whole route as unhashable if it has the hash key in it
-            if re.search(r'(^|[^^]){1}/', pattern):
+            if re.search(pattern, r'(^|[^^]){1}/'):
```

Comment:
I'm an idiot. `r'(^|[^^]){1}/'` is of course our regex pattern here. The `pattern` variable should be renamed since that is actually the string that to search. Also we definitely need a unit test for this since we shouldn't be able to switch these around.